File: python/uhflib/UHFReader.py
import serial
import time

class UHFReader:
    def __init__(self, com_port, baud_rate):
        self.ser = serial.Serial(com_port, baud_rate, timeout=2)

    def connect(self):
        if self.ser.is_open:
            print(f"Connected to {self.ser.port} at {self.ser.baudrate} baud.")

    def disconnect(self):
        self.ser.close()
        print("Disconnected from UHF Reader.")

    def read_tag(self):
        try:
            response = self.ser.readline().decode('utf-8').strip()
            return response
        except serial.SerialException as e:
            print(f"Error reading from the UHF reader: {e}")
            return None

    def set_power_level(self, dBm):
        command = f'SET_PWR {dBm}\n'.encode('utf-8')
        self.ser.write(command)

    def set_ping_rate(self, milliseconds):
        command = f'SET_PING {milliseconds}\n'.encode('utf-8')
        self.ser.write(command)

if __name__ == "__main__":
    reader = UHFReader('COM4', 115200)
    reader.connect()
    tag_data = reader.read_tag()
    print(f"Tag Data: {tag_data}")
    reader.disconnect()

File: php/uhflib/UHFReader.php
<?php

namespace Uhflib;

class UHFReader
{
    private $serialPort;

    public function __construct($comPort, $baudRate)
    {
        $this->serialPort = fopen($comPort, "w+b");

        if (!$this->serialPort) {
            throw new \Exception("Unable to open the COM port: $comPort");
        }

        stream_set_blocking($this->serialPort, 0);
        stream_set_timeout($this->serialPort, 2);

        // Assuming baud rate can be set by stty command
        exec("stty -F $comPort $baudRate");
    }

    public function connect()
    {
        if ($this->serialPort) {
            echo "Connected to UHF Reader.\n";
        }
    }

    public function disconnect()
    {
        if ($this->serialPort) {
            fclose($this->serialPort);
            echo "Disconnected from UHF Reader.\n";
        }
    }

    public function readTag()
    {
        if ($this->serialPort) {
            $data = stream_get_contents($this->serialPort);
            return trim($data);
        }
        return null;
    }

    public function setPowerLevel($dBm)
    {
        if ($this->serialPort) {
            $command = "SET_PWR $dBm\n";
            fwrite($this->serialPort, $command);
        }
    }

    public function setPingRate($milliseconds)
    {
        if ($this->serialPort) {
            $command = "SET_PING $milliseconds\n";
            fwrite($this->serialPort, $command);
        }
    }
}